youtube_dl/extractor/npo.py

```diff
@@ -82,686 +27,253 @@ class NPOIE(NPOBaseIE):
+                                        }).encode('utf8'),
+                                    headers={
+                                        'Content-Type': 'application/json',
+                                    })
+        product_id = media.get('data', {}).get('player', {}).get('pomsProductId')
+
+        formats = self._download_by_product_id(product_id, video_id)
 
         return {
-            'id': video_id,
-            'title': self._live_title(title) if is_live else title,
-            'description': metadata.get('info'),
-            'thumbnail': metadata.get('images', [{'url': None}])[-1]['url'],
-            'upload_date': unified_strdate(metadata.get('gidsdatum')),
-            'duration': parse_duration(metadata.get('tijdsduur')),
+            'id': product_id,
+            'title': media.get('data', {}).get('player', {}).get('title'),
             'formats': formats,
-            'subtitles': subtitles,
-            'is_live': is_live,
+            'thumbnail': media.get('data', {}).get('player', {}).get('image').get('url'),
         }
 
 
-class NPOLiveIE(NPOBaseIE):
-    IE_NAME = 'npo.nl:live'
-    _VALID_URL = r'https?://(?:www\.)?npo(?:start)?\.nl/live(?:/(?P<id>[^/?#&]+))?'
-
+class ONIE(NPOIE):
+    IE_NAME = 'on'
+    IE_DESC = 'ongehoordnederland.tv'
+    _VALID_URL = r'https?://(?:www\.)?ongehoordnederland.tv/.*'
     _TESTS = [{
-        'url': 'http://www.npo.nl/live/npo-1',
-        'info_dict': {
-            'id': 'LI_NL1_4188102',
-            'display_id': 'npo-1',
-            'ext': 'mp4',
-            'title': 're:^NPO 1 [0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}$',
-            'is_live': True,
-        },
-        'params': {
-            'skip_download': True,
-        }
-    }, {
-        'url': 'http://www.npo.nl/live',
-        'only_matching': True,
-    }, {
-        'url': 'https://www.npostart.nl/live/npo-1',
-        'only_matching': True,
+        'url': 'https://ongehoordnederland.tv/2024/03/01/korte-clips/heeft-preppen-zin-betwijfel-dat-je-daar-echt-iets-aan-zult-hebben-bij-oorlog-lydia-daniel/',
+        # TODO fill in other test attributes
     }]
 
     def _real_extract(self, url):
-        display_id = self._match_id(url) or 'npo-1'
-
-        webpage = self._download_webpage(url, display_id)
+        video_id = url.rstrip('/').split('/')[-1]
+        page, _ = self._download_webpage_handle(url, video_id)
+        results = re.findall("page: '(.+)'", page)
+        formats = []
+        for result in results:
+            formats.extend(self._download_by_product_id(result, video_id))
 
-        live_id = self._search_regex(
-            [r'media-id="([^"]+)"', r'data-prid="([^"]+)"'], webpage, 'live id')
+        if not formats:
+            raise ExtractorError('Could not find a POMS product id in the provided URL, '
+                                 'perhaps because all stream URLs are DRM protected.')
```

Comment:
`self._sort_formats(...)` should be called and will raise if there aren't any. If there is a way to identify that a stream has DRM, and given that unlike yt-dlp we're going to skip DRM formats, one could, eg, return a tuple '(formats, number_of formats_seen)` and then compare the sum of the total formats against `len(formats)`. If `not formats` and the values differ, `self.report_drm()` can be called.